src/main/scala/com/cloudera/sparkts/models/ARIMAX.scala

```diff
@@ -479,7 +479,7 @@ class ARIMAXModel(
     var error = 0.0
 
     val xregMatrix = new BreezeDenseMatrix(rows = exogenousVar.flatten.length / exogenousVar.size, cols = exogenousVar.size, data = exogenousVar.flatten)
-    val xregPredictors = AutoregressionX.assemblePredictors(ts.toArray, MatrixUtil.matToRowArrs(xregMatrix), 0, xregMaxLag, includeOriginalXreg)
+    val xregPredictors = AutoregressionX.assemblePredictors(Array.fill(xregMaxLag)(0.0) ++ ts.toArray, MatrixUtil.matToRowArrs(xregMatrix), 0, xregMaxLag, includeOriginalXreg)
```

Comment:
It doesnt change prediction results - in particular circumstances (when we have d > 0 and xMaxLag > 1) the length of timeseries values was bigger than predictors.